crates/db_views/src/comment/comment_view.rs

```diff
@@ -327,24 +316,8 @@ impl CommentQuery<'_> {
       .load::<CommentView>(conn)
       .await?;
 
-    let is_admin = o.local_user.map(|u| u.admin).unwrap_or(false);
-
-    // Note: deleted and removed comments are done on the front side
-    Ok(
-      res
-        .into_iter()
-        .map(|c| handle_deleted(c, is_admin))
-        .collect(),
-    )
-  }
-}
-
-/// Only show deleted / removed content for admins.
-fn handle_deleted(mut c: CommentView, is_admin: bool) -> CommentView {
-  if !is_admin && (c.comment.deleted || c.comment.removed) {
-    c.comment.content = String::new();
+    Ok(res)
```

Comment:
Instead of sql you could leave the logic in Rust like this:

```rust
if !c.can_mod && (c.comment.deleted || c.comment.removed) {
    c.comment.content = String::new();
}
c

Reply:
The SQL option is probably fine (and we should prefer SQL selects over post-select manipulation), but we can revisit if it ends up being a performance issue.